### PR TITLE
ENH: make too many/too few empty lines more verbose

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

**Comment by jakirkham** [comment]
@goanpeca , any thoughts on this tweak to the linter's end of line check?

**Comment by goanpeca** [comment]
ehhh sure why not. but tests should be fixed, no?

**Comment by jakirkham** [comment]
Yep, you're right. It will need test updates and possibly more tests to ensure we are covering both branches (though I thought we covered both cases before).

**Comment by CJ-Wright** [comment]
I guess I had ran into the case where I had too many lines and didn't know it and was a bit confused by the message for a little while. The idea was that we have the data needed to make a more verbose comment and it doesn't cost us much to write it out so we might as well be more verbose, it could save someone the confusion over of they have too few or too many lines. In retrospect we could even output exactly how many extra/fewer lines they have, although that was not in this PR.

**Comment by goanpeca** [comment]
> In retrospect we could even output exactly how many extra/fewer lines they have, although that was not in this PR.

I did not pursue this originally (I introduced the change you are improving) because of the way we run the test cases (we expect a fixed message in the tests not a variable one...) but I guess we can tweak the tests a bit to make this work.

**Comment by jakirkham** [comment]
Yep, I'm ok with this sort of change. Just need to update the tests.

**Comment by goanpeca** [comment]
@jakirkham this way of running tests is turning a bit cumbersome. We should make these messages warning classes so it is easier to test.

Thoughts?

**Comment by jakirkham** [comment]
Seems like a good idea. Could you please add it to a new issue?

**Comment by CJ-Wright** [comment]
If you are using pytest you might also look at [pytest.mark.parametrize ](http://doc.pytest.org/en/latest/parametrize.html)

**Comment by goanpeca** [comment]
@CJ-Wright, what I meant is that we can no longer user messages and test based on that. We need to move to warnings/exceptions. So this is not a refactor for the tests only, but for how the code is written.

**Comment by jakirkham** [comment]
Sort of tangential, PR ( https://github.com/conda-forge/conda-smithy/pull/385 ) would switch us to `pytest`. Currently we are using `unittest`.

**Comment by goanpeca** [comment]
cool I really prefer pytest over unittest

File: conda_smithy/tests/test_lint_recipe.py [hunk 1]
```diff
@@ -233,16 +233,25 @@ def test_end_empty_line(self):
...
+                                        'at the end of the '
+                                        'file.'.format(lines - 1))
+                else:
+                    expected_message = ('There are too few lines.  '
+                                        'There should be one empty line at'
+                                        ' the end of the file.')
                 if content == valid_content:
                     self.assertNotIn(expected_message, lints)
                 else:
+
```
Comment (jakirkham) [inline conda_smithy/tests/test_lint_recipe.py]: Can we drop this empty line?

**Comment by jakirkham** [comment]
Excepting one small nit. This looks fine to me.

@goanpeca, do you have any more thoughts?

**Comment by goanpeca** [comment]
LGTM :+1:

**Comment by jakirkham** [comment]
Once the nit is fixed, @CJ-Wright , will happily merge.

**Comment by jakirkham** [comment]
Thanks @CJ-Wright. This won't be in `2.0.0`, but it will be in whatever release we do next.

**Comment by jakirkham** [comment]
Thanks @CJ-Wright. This will be in `2.0.0`, but it will be in whatever release we do next.

**Comment by jakirkham** [comment]
Thanks @CJ-Wright. This will be in `2.0.0`.

**Comment by jakirkham** [comment]
Have released this to the linter webservice. So we should start seeing warnings for issues like this.

**Comment by CJ-Wright** [comment]
Thank you @jakirkham !

**Pull request merged**